flambe/field/text.py

```diff
@@ -90,10 +130,10 @@ def __init__(self,  # nosec
                  unk_token: str = '<unk>',
                  sos_token: Optional[str] = None,
                  eos_token: Optional[str] = None,
+                 embeddings_info: Optional[EmbeddingsInformation] = None,
```

Comment:
Why did you add this? Seems unnecessary, and adds an extra level of nesting

Reply:
By having all related parameters in the same data structure, the code becomes:

* more clear and readable, following a composition pattern allowing the creation of factories 
* easier to test
* easier to extend the class as your __init__ doesn't have tons of parameters anymore
* easier to implement custom logic in this new class rather than in the TextField iteself. For example, avoiding having in the TextField checks like "user passed the `embedding` name but `embedding_format` is None" which can be delegated to the appropriate class.

Users won't have another level of nesting in they configs as they will use the factory

Reply:
I think is a great idea to take subclassing and logic extraction into appropiate components 👏